src/app/app.module.ts

```diff
@@ -91,68 +94,93 @@ export function appInitializer(
 
           // Set this back so that second flow through we do the proper code flow to get a refresh token.
           sessionStorage.setItem('syndesis-first-idp', 'false');
-          oauthService.hybrid = originalHybrid;
 
-          let autoLinkGithHub = configService.getSettings('oauth')[
-            'auto-link-github'
-          ];
-          if (autoLinkGithHub === undefined) {
-            autoLinkGithHub = true;
-          }
+          /**
+           * Redirects user to setup page if GitHub setup is still pending
+           */
+          const apiEndpoint = configService.getSettings().apiEndpoint;
+          const accessToken = oauthService.getAccessToken();
+          setupService.isSetupPending(apiEndpoint, accessToken)
+            .then(function (setupPending) {
+              if (setupPending) {
+                window.location.assign('/setup');
```

Comment:
This feels like it could be a bit brittle, would it be better to set an attribute on setupService and then do this from the app component's `ngOnInit` function using `router.navigate()` instead?

Reply:
I agree. I can try to work on this in the morning, way too late now.